src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,113 @@
+package wolox.training.controllers;
+
+import static java.util.Optional.ofNullable;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.content;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.ArrayList;
+import java.util.List;
+import javax.annotation.PostConstruct;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(BookController.class)
+public class BookControllerTest {
+    @Autowired
+    private MockMvc mvc;
+
+    private Book bookTest;
+    private List<Book> books;
+
+    @MockBean
+    private BookRepository mockBookRepository;
+
+    private ObjectMapper objectMapper;
+
+    @PostConstruct
+    private void objectMapper() {
+        objectMapper = new ObjectMapper();
+    }
+
+    @Before
+    public void setUp() {
+        bookTest = new Book();
+        bookTest.setAuthor("DC: Alan Moore");
+        bookTest.setImage("image.png");
+        bookTest.setTitle("Batman: The Killing Joke");
+        bookTest.setSubtitle("A subtitle");
+        bookTest.setPublisher("Publisher");
+        bookTest.setGenre("novela grafica");
+        bookTest.setYear("1988");
+        bookTest.setPages(100);
+        bookTest.setIsbn("ABCD1234");
+        books = new ArrayList<>();
+
+    }
+
+    @Test
+    public void whenGetAllBooks_thenReturnAListOfBooks() throws Exception {
+        books.add(bookTest);
+        Mockito.when(mockBookRepository.findAll()).thenReturn(books);
+        mvc.perform(get("/api/books/").contentType(MediaType.APPLICATION_JSON))
```

Comment:
You can extract this URL path into a constant
```suggestion
        mvc.perform(get(BOOKS_PATH).contentType(MediaType.APPLICATION_JSON))
```